### PR TITLE
Feature 13348 Support for admin freeze and unfreeze

### PR DESCRIPTION
**Pull request by ritvick**
*Issue #:* /bcgov/entity/issues/13348

*Description of changes:*
Front end changes for Freeze and Unfreeze

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/store/getters.ts
Comment (severinbeauvais) [inline src/store/getters.ts:213]: Remove debugging code.

File: src/enums/filingNames.ts
Comment (severinbeauvais) [inline src/enums/filingNames.ts:23]: Please sort alphabetically to make it easier to look for a key in the future.

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:196]: Please rename this to be consistent with the other similar getters.

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:370]: Which code block builds the filing body for this new filing? If it is the block on lines 348-355 then please update the comment there.

File: src/store/getters.ts
Comment (severinbeauvais) [inline src/store/getters.ts:214]: I don't see in this PR where this store value is declared, initialized, or set in any way. Are some files missing from this PR?

File: src/enums/filingTypes.ts
Comment (severinbeauvais) [inline src/enums/filingTypes.ts:23]: Please sort alphabetically to make it easier to look for a key in the future.

File: src/mixins/enum-mixin.ts
Comment (severinbeauvais) [inline src/mixins/enum-mixin.ts:162]: You should update the method on line 236 (and its unit test).

Also, is any special needed to display this filing in the Recent Filing History?

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:24]: You might as well use [the Vuetify helper class](https://vuetifyjs.com/en/styles/text-and-typography/) to make the text bold.

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [reply src/components/dialogs/AddStaffNotationDialog.vue:370]: Also, I'm thinking there's a property missing in the filing to indicate whether it is Freezing or Unfreezing the business.

File: src/components/Dashboard/StaffNotation.vue
Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:101]: So this item is not displayed if the business is not frozen? Then why does the dialog support both freezing and unfreezing?

Comment (severinbeauvais) [reply src/components/Dashboard/StaffNotation.vue:101]: Also, after you rebase, please update your new code to match the other templates/v-list-items. One notable change is that the v-if on line 101 should be moved to line 100 (so we don't get an empty template if the condition is false).

**Review by severinbeauvais — CHANGES_REQUESTED** [review]
1. Please rebase and update your code to match any similar changes.
2. Please see my other comments. It looks like there are unfinished things.
3. Please update the app version and check in both package files.

File: src/store/getters.ts
Comment (ritvick) [reply src/store/getters.ts:214]: It is being set here https://github.com/bcgov/business-filings-ui/blob/main/src/store/mutations.ts#L30

File: src/mixins/enum-mixin.ts
Comment (ritvick) [reply src/mixins/enum-mixin.ts:168]: image

Comment (ritvick) [reply src/mixins/enum-mixin.ts:168]: this will appear like this ^ above screenshot

**Comment by ChrisGabel23** [comment]
Looks like two SonarCloud Bugs have been introduced in this PR.

**Comment by ritvick** [comment]
> Looks like two SonarCloud Bugs have been introduced in this PR.

Strange, the two bugs are in the file FrozenInformation.vue, which is exactly copied from MissingInformation.vue

File: src/store/getters.ts
Comment (severinbeauvais) [reply src/store/getters.ts:210]: OK, I found it. I did not realize this state variable existed previously and I expected it to be declared and initialized in this PR.

Declared: https://github.com/bcgov/business-filings-ui/blob/61870a728eef653a169c178170d0a7f3fb0c4d02/src/interfaces/state-interface.ts#L20

Initialized: https://github.com/bcgov/business-filings-ui/blob/61870a728eef653a169c178170d0a7f3fb0c4d02/src/store/state.ts#L14

I'm resolving this conversation.

File: src/mixins/enum-mixin.ts
Comment (severinbeauvais) [reply src/mixins/enum-mixin.ts:168]: Thanks! A picture is worth 1000 words. This shows the new alert and the ledger item in the history 👍 

However, I notice here that the Business Address Change and Proprietor Change buttons are still enabled, but they should be disabled.

File: src/components/Dashboard/StaffNotation.vue
Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:142]: "freeze" should probably be capitalized.

File: src/components/Dashboard/StaffNotation.vue
Comment (severinbeauvais) [inline src/components/Dashboard/StaffNotation.vue:140]: Please format this line like the others above and add the data-type property.

Also please update the unit tests for this component.

File: src/components/dialogs/AddStaffNotationDialog.vue
Comment (severinbeauvais) [inline src/components/dialogs/AddStaffNotationDialog.vue:24]: This line is longer than the 120 limit. If possible, keep it about the same length as the other lines in this component.

image

**Comment by severinbeauvais** [comment]
> > Looks like two SonarCloud Bugs have been introduced in this PR.
> 
> Strange, the two bugs are in the file FrozenInformation.vue, which is exactly copied from MissingInformation.vue

You can click on the Details link to see more info. These are SonarCloud false positives. Are you able to login to SonarCloud to resolve these? If not, ask me and I will do it for you.

**Comment by severinbeauvais** [comment]
I will approve this after the fixes are made and after some other reviewers approve. (It is not right for me to be the sole reviewer for PRs.)

File: tests/unit/StaffNotation.spec.ts
Comment (severinbeauvais) [inline tests/unit/StaffNotation.spec.ts:89]: If business is historical (as in this test), should this say Administer Unfreeze instead? Please confirm with UX team.

File: tests/unit/StaffNotation.spec.ts
Comment (severinbeauvais) [inline tests/unit/StaffNotation.spec.ts:229]: "Truthy" is almost anything. It's better to check `toBe(true)`, which is an exact match.

(To all devs: same for "toBeFalsy": could be 0, false, empty string, null, undefined. That is not very specific. We should be specific since we should know exactly what the value is.)

File: src/mixins/enum-mixin.ts
Comment (severinbeauvais) [reply src/mixins/enum-mixin.ts:168]: Did updating the "blocker" getter fix the Business Address Change and Proprietor Change buttons?

File: tests/unit/StaffNotation.spec.ts
Comment (ritvick) [reply tests/unit/StaffNotation.spec.ts:89]: If the business is historical, Administer Freeze / Unfreeze will not appear as an option.

Comment (severinbeauvais) [reply tests/unit/StaffNotation.spec.ts:89]: OK, I understand.

**Review by severinbeauvais — COMMENTED** [review]
There is 1 outstanding ~discussion~ conversation.

Also, I will approve after other reviewers have a look.

**Review by JazzarKarim — APPROVED** [review]

**Review by ChrisGabel23 — APPROVED** [review]

**Comment by severinbeauvais** [comment]
There is still an unresolved conversation and the branch is out of date with main.

File: src/mixins/enum-mixin.ts
Comment (seeker25) [reply src/mixins/enum-mixin.ts:168]: Yes it did Severin:

image

File: src/components/Dashboard/Alerts/FrozenInformation.vue
Comment (seeker25) [inline src/components/Dashboard/Alerts/FrozenInformation.vue:10]: Not sure why we're using integer instead of boolean?

File: src/components/Dashboard/Alerts/FrozenInformation.vue
Comment (seeker25) [inline src/components/Dashboard/Alerts/FrozenInformation.vue:50]: `this.panel = !this.panel` if it was boolean

File: src/components/Dashboard/Alerts/FrozenInformation.vue
Comment (seeker25) [inline src/components/Dashboard/Alerts/FrozenInformation.vue:55]: Confirmed this is in LD

**Comment by seeker25** [comment]
It's up to date with main, but there seem to be some unrelated unit tests failing.

**Comment by severinbeauvais** [comment]
Are staff still able to change non-structural data, such as comments on the business or filings?

Re: line 45 in [Business States RFC](https://github.com/bcgov/entity/blob/main/rfcs/rfc-business-states.md)

If the above is out of scope for the current ticket/PR then please create a new (bug?) ticket. Thx.

**Comment by severinbeauvais** [comment]
PS The units are failing because the enum values changed. @ketaki-deodhar is fixing them in her PR so I think it's OK to ignore them here.

File: src/components/Dashboard/Alerts/FrozenInformation.vue
Comment (severinbeauvais) [reply src/components/Dashboard/Alerts/FrozenInformation.vue:10]: Because it's the panel number -- see line 2.

File: src/components/Dashboard/Alerts/FrozenInformation.vue
Comment (severinbeauvais) [reply src/components/Dashboard/Alerts/FrozenInformation.vue:50]: Can't be a boolean.

**Comment by seeker25** [comment]
Can still add comments to filings, comments to the business.  Staff can't change business addresses or proprietor when frozen though.

**Pull request merged**
